packages/venia-concept/src/components/Pagination/pagination.js

```diff
@@ -171,28 +166,31 @@ class Pagination extends Component {
         const { location, pageControl } = this.props;
         const { currentPage, setPage, totalPages } = pageControl;
 
-        const queryPage = Math.max(
-            1,
-            // Note: The ~ operator is a bitwise NOT operator.
-            // Bitwise NOTing any number x yields -(x + 1). For example, ~-5 yields 4.
-            // Importantly, it truncates any fractional component of x. For example, ~-5.7 also yields 4.
-            // For positive numbers, applying this operator twice has the same effect as Math.floor.
-            ~~getQueryParameterValue({
-                location,
-                queryParameter: 'page'
-            })
-        );
+        // Indicator that our data has been loaded and we know enough to bounds check
+        if (totalPages) {
```

Comment:
What is the case in which we would be loaded but not have `totalPages`?

Reply:
Somehow a render of pagination is entering the event loop where `usePagination` is still in it's initialState (`totalPages = null`), but this is a very good question. Persistent pagination logic in `category.js` might be conflicting with this queryParams logic. With the assumption that the pagination would go through its own re-factor, I was going for a quick fix, but I can look into it more.

Reply:
Fixed by f209db6; digging into it, the child component previously never rendered until totalPages was set, which I've now restored by using that as our loading check.